src/state.rs

```diff
@@ -0,0 +1,106 @@
+use anyhow::{anyhow, Result};
+use std::{
+    collections::HashMap,
+    hash::Hash,
+    net::TcpStream,
+    sync::{Arc, Mutex},
+};
+
+use crate::{pipeline::Pipeline, tcp_connections};
+
+#[allow(dead_code)]
+pub struct Frame {
+    pub data: bytes::Bytes,
+}
+
+#[allow(dead_code)]
+pub struct Input {
+    port: u16,
+    input_id: u32,
+}
+
+#[allow(dead_code)]
+pub struct Output {
+    port: u16,
+    output_id: u32,
+}
+
+#[allow(dead_code)]
+pub struct PendingConnection {
+    pub port: u16,
+    pub tcp_stream: TcpStream,
+}
+
+pub struct SyncHashMap<K, V>(std::sync::Mutex<HashMap<K, V>>);
+
+impl<K, V> SyncHashMap<K, V>
```

Comment:
We could use some lib, but here we are dealing with very small maps, so this approach is good enough, and this way we can customize it more for our needs.